jobs/scrap_archives.py

```diff
@@ -0,0 +1,190 @@
+    return await asyncio.gather(*futures)
+
+
+async def fetch_versions(session, queue, product):
+    product_url = archive_url(product)
+    versions_folders, _ = await fetch_listing(session, product_url)
+
+    versions = [v[:-1] for v in versions_folders if re.match(r'^[0-9]', v)]
+    versions = [v for v in versions if "funnelcake" not in v]
+    versions = sorted(versions, key=lambda v: version_parse(v), reverse=True)
+
+    futures = [fetch_platforms(session, queue, product, version)
+               for version in versions]
+    return await asyncio.gather(*futures)
+
+
+async def fetch_platforms(session, queue, product, version):
+    version_url = archive_url(product, version)
+    platform_folders, _ = await fetch_listing(session, version_url)
+
+    platforms = [p[:-1] for p in platform_folders]
+    platforms = [p for p in platforms if p not in ("source", "update")]
+
+    futures = [fetch_locales(session, queue, product, version, platform)
+               for platform in platforms]
+    return await asyncio.gather(*futures)
+
+
+async def fetch_locales(session, queue, product, version, platform):
+    platform_url = archive_url(product, version, platform)
+    locale_folders, _ = await fetch_listing(session, platform_url)
+
+    locales = [l[:-1] for l in locale_folders]
+    locales = [l for l in locales if l != "xpi"]
+
+    futures = [fetch_files(session, queue, product, version, platform, locale)
+               for locale in locales]
+    return await asyncio.gather(*futures)
+
+
+async def fetch_files(session, queue, product, version, platform, locale):
+    locale_url = archive_url(product, version, platform, locale)
+    _, files = await fetch_listing(session, locale_url)
+
+    fileregexp = re.compile("%s-(.+)(zip|gz|bz|bz2|dmg|apk)$" % product)
+    files = [f for f in files if fileregexp.match(f["name"])]
+
+    futures = []
+    for file_ in files:
+        filename = file_["name"]
+        record = {
+            "build": None,
+            "source": {
+                "revision": None,
+                "tree": None,
+                "product": product,
+            },
+            "target": {
+                "platform": platform,
+                "locale": locale,
+                "version": version,
+                "channel": None,
+            },
+            "download": {
+                "url": locale_url + filename,
+                "mimetype": None,
+                "size": None,
+            },
+            "systemaddons": None
+        }
+        logger.debug("Release found %s" % record["download"]["url"])
+        futures.append(queue.put(record))
+
+    return await asyncio.gather(*futures)
+
+
+async def produce(loop, queue):
+    async with aiohttp.ClientSession(loop=loop) as session:
+        await fetch_products(session, queue, PRODUCTS)
+    logger.info("Scraping releases done.")
+
+
+async def consume(loop, executor, client, queue):
```

Comment:
Add a docstring `"""Store grabbed releases from the archives website in Kinto."""`